schematic/schemas/generator.py

```diff
@@ -105,6 +106,31 @@ def get_adjacent_nodes_by_relationship(self,
         return sorted(list(nodes))
 
 
+    def get_subgraph_by_edge_type(self,
+                                  graph: nx.MultiDiGraph,
+                                  relationship:str) -> nx.DiGraph:
```

Comment:
I noticed that `self` isn't used in the function body. Based on future use cases, does it make more sense to use `self.se.schema_nx` (or some other `self` attribute) within the function instead of the `graph` argument or to make this function a standalone utility function? Alternatively, we could drop the `self` argument by making this a [static method](https://docs.python.org/3/library/functions.html#staticmethod) rather than an instance method. I defer to @milen-sage to determine which approach is ideal.

Reply:
good point @BrunoGrandePhD 

I'm leaning towards dropping the self attribute and going static short-term (we shouldn't assume we always work with the full schema graph; e.g. we could pass a subgraph of the schema). Longer-term the "generator" class will likely need to be refactored; at that point, there will likely be some methods that go into utils.